sdcm/utils/health_checker.py

```diff
@@ -91,39 +91,40 @@ def check_node_status_in_gossip_and_nodetool_status(gossip_info, nodes_status, c
-        if ip not in nodes_status:
+    for node, node_info in gossip_info.items():
+        is_target = current_node.print_node_running_nemesis(node)
+        if node not in nodes_status:
             if node_info['status'] not in current_node.GOSSIP_STATUSES_FILTER_OUT:
                 LOGGER.debug("Gossip info: %s\nnodetool.status info: %s", gossip_info, nodes_status)
                 yield ClusterHealthValidatorEvent.NodeStatus(
                     severity=Severity.ERROR,
                     node=current_node.name,
-                    error=f"Current node {current_node.ip_address}. The node {ip}{is_target} exists in the gossip but "
-                          f"doesn't exist in the nodetool.status",
+                    error=f"Current node {current_node.ip_address}. The node {node.ip_address}{is_target} "
+                          f"exists in the gossip but doesn't exist in the nodetool.status",
                 )
             continue
 
-        if (node_info['status'] == 'NORMAL' and nodes_status[ip]['status'] != 'UN') or \
-                (node_info['status'] != 'NORMAL' and nodes_status[ip]['status'] == 'UN'):
+        if (node_info['status'] == 'NORMAL' and nodes_status[node]['status'] != 'UN') or \
+                (node_info['status'] != 'NORMAL' and nodes_status[node]['status'] == 'UN'):
             yield ClusterHealthValidatorEvent.NodeStatus(
                 severity=Severity.ERROR,
                 node=current_node.name,
-                error=f"Current node {current_node.ip_address}. Wrong node status. Node {ip}{is_target} status in "
-                      f"nodetool.status is {nodes_status[ip]['status']}, but status in gossip {node_info['status']}",
+                error=f"Current node {current_node.ip_address}. Wrong node status. "
+                      f"Node {node.ip_address}{is_target} status in nodetool.status is "
+                      f"{nodes_status[node]['status']}, but status in gossip {node_info['status']}",
             )
 
     # Validate that all nodes in nodetool.status exist in gossip
     not_in_gossip = list(set(nodes_status.keys()) - set(gossip_info.keys()))
-    for ip in not_in_gossip:
-        if nodes_status[ip]['status'] == 'UN':
-            is_target = current_node.print_node_running_nemesis(ip)
+    for node in not_in_gossip:
+        if nodes_status[node]['status'] == 'UN':
+            is_target = current_node.print_node_running_nemesis(node.ip_address)
             LOGGER.debug("Gossip info: %s\nnodetool.status info: %s", gossip_info, nodes_status)
             yield ClusterHealthValidatorEvent.NodeSchemaVersion(
                 severity=Severity.ERROR,
                 node=current_node.name,
                 error=f"Current node {current_node.ip_address}. "
-                      f"Node {ip}{is_target} exists in the nodetool.status but missed in gossip.",
+                      f"Node {node.ip_address}{is_target} exists in the nodetool.status but missed in gossip.",
```

Comment:
better to use node, it has __str__ defined

Reply:
done